### PR TITLE
Closing file after uploading finished in UploadFile

### PR DESCRIPTION
**Pull request by somedev7**
I needed to remove just uploaded file and when I tried to remove file I got error "file in use". This fix resolved my problem.

### CODE REVIEW

File: mega.go [hunk 1]
```diff
@@ -1617,6 +1617,11 @@ func (m *Mega) UploadFile(srcpath string, parent *Node, name string, progress *c
 		return nil, err
 	}
 
+	err = infile.Close()
```
Comment (ncw) [inline mega.go:1620]: That would be better done in a `defer func() { ... }` just after the `OpenFile` I think

Comment (somedev7) [reply mega.go:1620]: but how to return error in defer?

Comment (ncw) [reply mega.go:1620]: Name the return parameters

```go
func (m *Mega) UploadFile(srcpath string, parent *Node, name string, progress *chan int) (node *Node, err error) {
```

Then you can refer to `err` in the `defer func()` something like this which takes care not to overwrite err if it is already set.

```go
defer func() {
    closeErr := infile.Close()
    if closeErr != nil && err == nil {
        err = closeErr
    }
}()
```

Comment (somedev7) [reply mega.go:1620]: then error with closing could be lost if there will be error in return statement, but in current case it's important to call u.Finish(), so I refactored my fix, thank you

**Comment by ncw** [comment]
That looks great now - thank you very much :-)

I'll merge it now.

**Pull request merged**